dialog/file.go

```diff
@@ -20,22 +20,72 @@ type textWidget interface {
 }
 
 type fileDialog struct {
+	file       *FileDialog
 	fileName   textWidget
 	open       *widget.Button
 	breadcrumb *widget.Box
 	files      *fyne.Container
 	fileScroll *widget.ScrollContainer
-	parent     fyne.Window
 
 	win      *widget.PopUp
 	selected *fileDialogItem
-	callback interface{}
 	dir      string
+}
+
+// FileDialog is a dialog containing a file picker for use in opening or saving files.
+type FileDialog struct {
 	save     bool
+	callback interface{}
```

Comment:
I wonder if we should have openCallback and saveCallback to avoid the type casts later on

Reply:
I don't mind the type casts later, but perhaps others may have some thought here.

Reply:
Unresolving this to see if others have an opinion here

Reply:
I didn't see that this had been reopened, sorry for that.
I agree that it could be improved, it has evolved a little since the first commit and we learned a lot along the way. Let's do a follow-on following discussion in Slack